File: algorithms/1_easy/C/02180_count_integers_with_even_digit_sum/solution.cpp
class Solution {
public:
    int countEven(int num) {
        int count {0};

        for (int i{1}; i <= num; ++i) {
            int temp {i};
            int dig_sum {0};

            while (temp > 0) {
                dig_sum += temp % 10;
                temp /= 10;
            }

            if ((dig_sum & 1) == 0) {
                count += 1;
            }
        }

        return count;
    }
};

File: algorithms/1_easy/C/02180_count_integers_with_even_digit_sum/solution.rs
impl Solution {
    pub fn count_even(num: i32) -> i32 {
        let mut count = 0_i32;

        for i in 1..=num {
            let mut temp = i;
            let mut dig_sum = 0;

            while temp > 0 {
                dig_sum += temp % 10;
                temp /= 10;
            }

            if (dig_sum & 1) == 0 {
                count += 1;
            }
        }

        count
    }
}
